### ISSUE TITLE
Lua scripts not executing atomically

### ISSUE THREAD

**Issue by mcmathja** [alicebob/miniredis#134]
Thanks for this library, it was very helpful during the development of [CurlyQ](https://github.com/mcmathja/curlyq).

I've run into an issue when running Lua scripts alongside blocking commands. I'm running against v2.11.1 of the library. My expectation according to the Redis spec is that Lua scripts should always execute atomically: if I push to a list that is the target of a blocking command inside of a Lua script, the Lua script should finish executing its remaining commands before the blocking operation proceeds. However, when using miniredis, it appears that the blocked operation will proceed immediately without waiting for the script to finish.

A quick repro:

```go
package main

import (
	"log"
	"time"

	"github.com/alicebob/miniredis/v2"
	"github.com/go-redis/redis/v7"
)

func main() {
	server, _ := miniredis.Run()
	client := redis.NewClient(&redis.Options{
		Addr: server.Addr(),
	})

	go func() {
		client.BRPopLPush("signal", "done", 100 * time.Second).Err()
	}()

	script := redis.NewScript(`
		redis.call("LPUSH", "signal", "a")
		redis.call("LPUSH", "done", "b")
		return 1
	`)
	script.Run(client, nil).Err()

	completionOrder, _ := client.LRange("done", 0, -1).Result()
	log.Println(completionOrder)
}
```

This results in output `[b a]`. The expected output, and the result I get if I run the same test against a live Redis server, is `[a b]`. Converting the script to a multi/exec produces the correct output. Any thoughts on where I might start looking if I wanted to have a go at fixing this?

**Comment by mcmathja** [comment]
Never mind, I just realized this is fixed in master. Apologies for the stray issue.

**Comment by alicebob** [comment]
> Never mind, I just realized this is fixed in master. Apologies for the stray issue.

I should have made a new release earlier, sorry about that :(
I published v2.11.2, which has the fix.

Thanks for the report and the kind words!

**Comment by mcmathja** [comment]
No worries, thanks for the release!
